### ISSUE TITLE
[JENKINS-25869] Jenkins is getting crashed and shutting down once in a while

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#23646]
<p>Our Jenkins master instance got crashed around 3 times in one week and it is getting shutdown by itself once in a while(once or twice per a day). I had mentioned the Environment details in the Environment Field. I am also attaching the last three Jenkins crash logs. This is a serious blocker for our Team, Please let us know what is the issue and remedy.</p>

<p>Please do let me know if you need further details</p>

<p>Quick response will be much useful !! Thanks in advance</p>

<p>Regards,<br/>
Vamsi Hari</p>

---
<details><summary><i>Originally reported by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=hvamsik">hvamsik</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25869" target="_blank">Jenkins is getting crashed and shutting down once in a while</a></i></summary>
<i><ul>
<li><b>status</b>: Resolved
<li><b>priority</b>: Blocker
<li><b>component(s)</b>: core
<li><b>label(s)</b>: crash, jenkins
<li><b>resolution</b>: Not A Defect
<li><b>resolved</b>: 2014-12-03T02:09:16+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Our Jenkins master instance got crashed around 3 times in one week and it is getting shutdown by itself once in a while(once or twice per a day). I had mentioned the Environment details in the Environment Field. I am also attaching the last three Jenkins crash logs. This is a serious blocker for our Team, Please let us know what is the issue and remedy.</p>

<p>Please do let me know if you need further details</p>

<p>Quick response will be much useful !! Thanks in advance</p>

<p>Regards,
Vamsi Hari</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Jenkins version=1.580.1, OS=Linux, Flavor=SLES11, Java version=1.6 Container=Tomcat 7.0.47</code></li></ul>
<details><summary><i>9 attachments</i></summary>

- [PortalEXT-EAR-deployment-PEO-config.xml](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/28317/PortalEXT-EAR-deployment-PEO-config.xml)
- [PortalEXT-myts-static-STATIC-deployment-config.xml](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/28318/PortalEXT-myts-static-STATIC-deployment-config.xml)
- [System Information [Jenkins].htm](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/28320/System%20Information%20%5BJenkins%5D.htm)
- [hs_err_pid12702.log](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/28109/hs_err_pid12702.log)
- [hs_err_pid13387.log](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/28406/hs_err_pid13387.log)
- [hs_err_pid22195.log](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/28300/hs_err_pid22195.log)
- [hs_err_pid26226.log](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/28110/hs_err_pid26226.log)
- [hs_err_pid5705.log](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/28108/hs_err_pid5705.log)
- [support.zip](https://raw.githubusercontent.com/jenkinsci/attachments-from-jira-issues-core-cli/refs/heads/main/attachments/28319/support.zip)
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25869?focusedId=217197&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217197">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is a bug tracker, not a support site.</p>
</pre>
</details>

</ul>
</details>

<p>This is a bug tracker, not a support site.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25869?focusedId=217198&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217198">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>That said, I'd try to figure out why it crashes in the same proxy related code and investigate the proxy configuration in plugin manager, proxy related environment variables, and/or system properties.</p>
</pre>
</details>

</ul>
</details>

<p>That said, I'd try to figure out why it crashes in the same proxy related code and investigate the proxy configuration in plugin manager, proxy related environment variables, and/or system properties.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>hvamsik:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25869?focusedId=217223&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217223">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks a lot Daniel for the prompt response,
Adding to what you said, we are also facing another issue (<a href="https://issues.jenkins-ci.org/browse/JENKINS-25870" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-25870</a>) which is related to proxy settings not working from plugin manager, anything related? 
More Information: Recent config changes that we have done with our Jenkins master is we had added a couple of windows slaves to run selenium Tests on them and we already have a linux slave.</p>

<p>My apologies, if this issue is not related to this Tracker, we just want to corner the issue, It would be so grateful if you can help us in cornering the issue</p>

<p>Please let me know if you need further details,</p>

<p>Below is the part of Tomcat log on the day when earlier crash happened. Below errors are repetitively found in logs (for better understanding i had renamed the hostnames of windows and linux slaves as "windowsslave" &amp; "Linuxslave" )</p>

<p>Nov 19, 2014 2:35:23 PM jenkins.slaves.JnlpSlaveAgentProtocol$Handler$1 onClosed
WARNING: Channel reader thread: WindowsSlave for + WindowsSlave terminated
java.io.IOException: Unexpected termination of the channel
        at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)
Caused by: java.io.EOFException
        at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2576)
        at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1295)
        at java.io.ObjectInputStream.readObject(ObjectInputStream.java:349)
        at hudson.remoting.Command.readFrom(Command.java:92)
        at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:71)
        at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)
Nov 19, 2014 2:35:33 PM hudson.TcpSlaveAgentListener$ConnectionHandler run
INFO: Accepted connection #12 from /51.16.108.187:61593
Nov 19, 2014 2:35:34 PM hudson.TcpSlaveAgentListener$ConnectionHandler run
WARNING: Connection #12 failed
java.io.EOFException: unexpected stream termination
        at hudson.remoting.ClassicCommandTransport.create(ClassicCommandTransport.java:135)
        at hudson.remoting.Channel.&lt;init&gt;(Channel.java:406)
        at hudson.remoting.Channel.&lt;init&gt;(Channel.java:402)
        at hudson.remoting.Channel.&lt;init&gt;(Channel.java:363)
        at hudson.remoting.Channel.&lt;init&gt;(Channel.java:359)
        at hudson.slaves.SlaveComputer.setChannel(SlaveComputer.java:344)
        at jenkins.slaves.JnlpSlaveAgentProtocol$Handler.jnlpConnect(JnlpSlaveAgentProtocol.java:122)
        at jenkins.slaves.JnlpSlaveAgentProtocol2$Handler2.run(JnlpSlaveAgentProtocol2.java:99)
        at jenkins.slaves.JnlpSlaveAgentProtocol2.handle(JnlpSlaveAgentProtocol2.java:43)
        at hudson.TcpSlaveAgentListener$ConnectionHandler.run(TcpSlaveAgentListener.java:154)</p>

<p>Nov 19, 2014 2:39:45 PM hudson.model.Run execute
INFO: servicegateway-static-build-git-claimsource #520 main build action completed: SUCCESS</p>

<p>Nov 19, 2014 2:39:54 PM hudson.remoting.Channel$2 handle
SEVERE: Failed to execute command Pipe.EOF(272) (channel Linuxslave)
java.lang.NullPointerException
        at hudson.remoting.ProxyWriter$EOF.execute(ProxyWriter.java:177)
        at hudson.remoting.Channel$2.handle(Channel.java:461)
        at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:60)
Nov 19, 2014 2:39:54 PM hudson.remoting.Channel$2 handle
SEVERE: This command is created here
Command Pipe.EOF(272) created at
        at hudson.remoting.Command.&lt;init&gt;(Command.java:67)
        at hudson.remoting.Command.&lt;init&gt;(Command.java:50)
        at hudson.remoting.ProxyWriter$EOF.&lt;init&gt;(ProxyWriter.java:169)
        at hudson.remoting.ProxyWriter.close(ProxyWriter.java:124)
        at hudson.remoting.ProxyWriter.finalize(ProxyWriter.java:132)
        at java.lang.ref.Finalizer.invokeFinalizeMethod(Native Method)
        at java.lang.ref.Finalizer.runFinalizer(Finalizer.java:83)
        at java.lang.ref.Finalizer.access$100(Finalizer.java:14)
        at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:172)
Nov 19, 2014 2:40:22 PM hudson.remoting.Channel$2 handle
SEVERE: Failed to execute command Pipe.EOF(275) (channel Linuxslave)
java.lang.NullPointerException
        at hudson.remoting.ProxyWriter$EOF.execute(ProxyWriter.java:177)
        at hudson.remoting.Channel$2.handle(Channel.java:461)
        at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:60)</p>


<p>Regards,
Vamsi Hari</p>

</pre>
</details>

</ul>
</details>

<p>Thanks a lot Daniel for the prompt response,<br/>
Adding to what you said, we are also facing another issue (<a href="https://issues.jenkins-ci.org/browse/JENKINS-25870" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-25870</a>) which is related to proxy settings not working from plugin manager, anything related? <br/>
More Information: Recent config changes that we have done with our Jenkins master is we had added a couple of windows slaves to run selenium Tests on them and we already have a linux slave.</p>

<p>My apologies, if this issue is not related to this Tracker, we just want to corner the issue, It would be so grateful if you can help us in cornering the issue</p>

<p>Please let me know if you need further details,</p>

<p>Below is the part of Tomcat log on the day when earlier crash happened. Below errors are repetitively found in logs (for better understanding i had renamed the hostnames of windows and linux slaves as "windowsslave" & "Linuxslave" )</p>

<p>Nov 19, 2014 2:35:23 PM jenkins.slaves.JnlpSlaveAgentProtocol$Handler$1 onClosed<br/>
WARNING: Channel reader thread: WindowsSlave for + WindowsSlave terminated<br/>
java.io.IOException: Unexpected termination of the channel<br/>
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:50)<br/>
Caused by: java.io.EOFException<br/>
at java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2576)<br/>
at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1295)<br/>
at java.io.ObjectInputStream.readObject(ObjectInputStream.java:349)<br/>
at hudson.remoting.Command.readFrom(Command.java:92)<br/>
at hudson.remoting.ClassicCommandTransport.read(ClassicCommandTransport.java:71)<br/>
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:48)<br/>
Nov 19, 2014 2:35:33 PM hudson.TcpSlaveAgentListener$ConnectionHandler run<br/>
INFO: Accepted connection #12 from /51.16.108.187:61593<br/>
Nov 19, 2014 2:35:34 PM hudson.TcpSlaveAgentListener$ConnectionHandler run<br/>
WARNING: Connection #12 failed<br/>
java.io.EOFException: unexpected stream termination<br/>
at hudson.remoting.ClassicCommandTransport.create(ClassicCommandTransport.java:135)<br/>
at hudson.remoting.Channel.<init>(Channel.java:406)<br/>
at hudson.remoting.Channel.<init>(Channel.java:402)<br/>
at hudson.remoting.Channel.<init>(Channel.java:363)<br/>
at hudson.remoting.Channel.<init>(Channel.java:359)<br/>
at hudson.slaves.SlaveComputer.setChannel(SlaveComputer.java:344)<br/>
at jenkins.slaves.JnlpSlaveAgentProtocol$Handler.jnlpConnect(JnlpSlaveAgentProtocol.java:122)<br/>
at jenkins.slaves.JnlpSlaveAgentProtocol2$Handler2.run(JnlpSlaveAgentProtocol2.java:99)<br/>
at jenkins.slaves.JnlpSlaveAgentProtocol2.handle(JnlpSlaveAgentProtocol2.java:43)<br/>
at hudson.TcpSlaveAgentListener$ConnectionHandler.run(TcpSlaveAgentListener.java:154)</p>

<p>Nov 19, 2014 2:39:45 PM hudson.model.Run execute<br/>
INFO: servicegateway-static-build-git-claimsource #520 main build action completed: SUCCESS</p>

<p>Nov 19, 2014 2:39:54 PM hudson.remoting.Channel$2 handle<br/>
SEVERE: Failed to execute command Pipe.EOF(272) (channel Linuxslave)<br/>
java.lang.NullPointerException<br/>
at hudson.remoting.ProxyWriter$EOF.execute(ProxyWriter.java:177)<br/>
at hudson.remoting.Channel$2.handle(Channel.java:461)<br/>
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:60)<br/>
Nov 19, 2014 2:39:54 PM hudson.remoting.Channel$2 handle<br/>
SEVERE: This command is created here<br/>
Command Pipe.EOF(272) created at<br/>
at hudson.remoting.Command.<init>(Command.java:67)<br/>
at hudson.remoting.Command.<init>(Command.java:50)<br/>
at hudson.remoting.ProxyWriter$EOF.<init>(ProxyWriter.java:169)<br/>
at hudson.remoting.ProxyWriter.close(ProxyWriter.java:124)<br/>
at hudson.remoting.ProxyWriter.finalize(ProxyWriter.java:132)<br/>
at java.lang.ref.Finalizer.invokeFinalizeMethod(Native Method)<br/>
at java.lang.ref.Finalizer.runFinalizer(Finalizer.java:83)<br/>
at java.lang.ref.Finalizer.access$100(Finalizer.java:14)<br/>
at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:172)<br/>
Nov 19, 2014 2:40:22 PM hudson.remoting.Channel$2 handle<br/>
SEVERE: Failed to execute command Pipe.EOF(275) (channel Linuxslave)<br/>
java.lang.NullPointerException<br/>
at hudson.remoting.ProxyWriter$EOF.execute(ProxyWriter.java:177)<br/>
at hudson.remoting.Channel$2.handle(Channel.java:461)<br/>
at hudson.remoting.SynchronousCommandTransport$ReaderThread.run(SynchronousCommandTransport.java:60)</p>


<p>Regards,<br/>
Vamsi Hari</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25869?focusedId=217671&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-217671">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>That looks like a completely different issue. What is the relation of that to the earlier reported JRE crashes?</p>
</pre>
</details>

</ul>
</details>

<p>That looks like a completely different issue. What is the relation of that to the earlier reported JRE crashes?</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25869?focusedId=218674&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-218674">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I'd try to make sure to unset the <tt>http_proxy</tt> and related environment variables before starting the Jenkins process (or rather Tomcat in your case?), instead (if necessary) using proxy configuration in Jenkins (Manage Jenkins » Manage Plugins » Advanced). This may result in child processes (e.g. shell scripts and the tools they run) not inheriting the proxy configuration by default though.</p>

<p>The bug may well be in the gconf (Gnome) library or the JDK, as that's where the crash occurs.
<a href="http://hg.openjdk.java.net/jdk7u/jdk7u/jdk/file/54acd5cd0485/src/solaris/native/sun/net/spi/DefaultProxySelector.c#l155" class="external-link" target="_blank" rel="nofollow noopener">http://hg.openjdk.java.net/jdk7u/jdk7u/jdk/file/54acd5cd0485/src/solaris/native/sun/net/spi/DefaultProxySelector.c#l155</a></p>

<p>Please also note that the user and password to the proxy are included in plain text in the support bundle (nodes/master/environment.txt), so you should change those now.</p>
</pre>
</details>

</ul>
</details>

<p>I'd try to make sure to unset the <tt>http_proxy</tt> and related environment variables before starting the Jenkins process (or rather Tomcat in your case?), instead (if necessary) using proxy configuration in Jenkins (Manage Jenkins » Manage Plugins » Advanced). This may result in child processes (e.g. shell scripts and the tools they run) not inheriting the proxy configuration by default though.</p>

<p>The bug may well be in the gconf (Gnome) library or the JDK, as that's where the crash occurs.<br/>
<a href="http://hg.openjdk.java.net/jdk7u/jdk7u/jdk/file/54acd5cd0485/src/solaris/native/sun/net/spi/DefaultProxySelector.c#l155" class="external-link" target="_blank" rel="nofollow noopener">http://hg.openjdk.java.net/jdk7u/jdk7u/jdk/file/54acd5cd0485/src/solaris/native/sun/net/spi/DefaultProxySelector.c#l155</a></p>

<p>Please also note that the user and password to the proxy are included in plain text in the support bundle (nodes/master/environment.txt), so you should change those now.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>hvamsik:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-25869?focusedId=218755&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-218755">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Thanks Daniel for your help,
  I will try removing all the proxy related environment variables and restarting the Jenkins container(Tomcat) and let you know about the result.</p>
</pre>
</details>

</ul>
</details>

<p>Thanks Daniel for your help,<br/>
  I will try removing all the proxy related environment variables and restarting the Jenkins container(Tomcat) and let you know about the result.</p>
